sycl/test/basic_tests/aspect_usage.cpp

```diff
@@ -0,0 +1,50 @@
+// Test checks Propagate Aspect Usage pass.
+
+// RUN: %clangxx -fsycl -fno-legacy-pass-manager %s -emit-llvm -S -o - | FileCheck %s --check-prefix CHECK-NODES
+// RUN: %clangxx -fsycl -fno-legacy-pass-manager -fno-sycl-early-optimizations %s -emit-llvm -S -o - | FileCheck %s --check-prefix CHECK-NODES
+// RUN: %clangxx -fsycl -flegacy-pass-manager %s -emit-llvm -S -o - | FileCheck %s --check-prefix CHECK-NODES
+// RUN: %clangxx -fsycl -flegacy-pass-manager -fno-sycl-early-optimizations %s -emit-llvm -S -o - | FileCheck %s --check-prefix CHECK-NODES
+
+// fp16 aspect corresponds to constant 5.
+// CHECK-NODES-DAG: !intel_used_aspects ![[NODE:[0-9]+]]
+// CHECK-NODES: ![[NODE]] = !{i32 5}
+
+// RUN: %clangxx -fsycl -fno-legacy-pass-manager %s -emit-llvm -S -o /dev/null 2>&1 | FileCheck %s --check-prefix CHECK-WARNINGS
+// RUN: %clangxx -fsycl -flegacy-pass-manager %s -emit-llvm -S -o /dev/null 2>&1 | FileCheck %s --check-prefix CHECK-WARNINGS
+
+// RUN: %clangxx -fsycl -fno-legacy-pass-manager %s -g -emit-llvm -S -o /dev/null 2>&1 | FileCheck %s --check-prefix CHECK-WARNINGS-DBG -DPATH=%s
+// RUN: %clangxx -fsycl -flegacy-pass-manager %s -g -emit-llvm -S -o /dev/null 2>&1 | FileCheck %s --check-prefix CHECK-WARNINGS-DBG -DPATH=%s
+
+// CHECK-WARNINGS: warning: function '_Z5func1v' uses aspect '5' not listed in `sycl::device_has()`
+// CHECK-WARNINGS-NEXT:  _Z5func1v()
+// CHECK-WARNINGS-NEXT:  _Z5func2v()
+// CHECK-WARNINGS-NEXT:  _Z5func3v()
+// CHECK-WARNINGS-NEXT: compile with '-g' to get source location
+
+// CHECK-WARNINGS-DBG: warning: function '_Z5func1v' uses aspect '5' not listed in `sycl::device_has()`
+// CHECK-WARNINGS-DBG-NEXT:  _Z5func1v() [[PATH]]:45:57
+// CHECK-WARNINGS-DBG-NEXT:  _Z5func2v() [[PATH]]:43:22
+// CHECK-WARNINGS-DBG-NEXT:  _Z5func3v() [[PATH]]:38:10
```

Comment:
Can we improve the call stack message here? I mean that it would be great to give user more info about what those names mean, something like:

```
in function _Z5func1v (defined at [[PATH]]:45:57)
which calls _Z5func2v (defined at [[PATH]]:43:22)
which calls _Z5func3v (defined at [[PATH]]:38:10)
which uses aspect '5'
```

Or otherwise:
```
aspect 5 is used in function _Z5func3v (defined at [[PATH]]:38:10)
which is called by _Z5func2v (defined at [[PATH]]:43:22)
which is called by _Z5func1v (defined at [[PATH]]:45:57)
```